_posts/2022-01-28-filesystem-encryption-integrity.adoc

```diff
@@ -0,0 +1,145 @@
+
+== An overview of the new attributes
+
+The filesystem realm now supports the following attributes
+
+* ``credential-store``: This attribute specifies the credential store resource where the secret key resides in. This credential store must be configured prior to creating the filesystem realm. This attribute is optional and if not specified, the filesystem realm will not be encrypted.
+
+* ``secret-key``: This attribute specifies the alias to the secret key in the credential store to encrypt the relam. By default the secret key alias is set to `key` when creating the credential store. This attribute is required if the `credential-store` attribute is specified.
+
+* ``key-store``: This attribute specifies the key store resource where the key pair resides in. This key store must be configured prior to creating the filesystem realm. This attribute is optional and if not specified, the filesystem realm will not verify it's integrity.
+
+* ``key-store-alias``: This attribute specifies the alias to the key pair in the key store to verify filesystem integrity. This attribute is required if the `key-store` attribute is specified.
+
+== Reviewing the Elytron Configuration
+
+In the following section, we will review the configuration available in the script for the quickstart
+*https://github.com/wildfly-security-incubator/elytron-examples/blob/master/encryption-integrity-filesystem-realm/configure-elytron.cli[configure-elytron.cli]*. We start our configuration by connecting to the server using the following command:
+
+[source,shell]
+----
+$ WILDFLY_HOME/bin/jboss-cli.sh --connect
+----
+==== Note: Use of WILDFLY_HOME
+In the following post, replace ``WILDFLY_HOME`` with the actual path to your WildFly installation.
+
+We first create a keystore and keypair under the Elytron subsystem, with the name ``keystore``, and alias ``user``.
+[source]
+----
+/subsystem=elytron/key-store=keystore:add(path=keystore, relative-to=jboss.server.config.dir, type=JKS, credential-reference={clear-text=secret})
+/subsystem=elytron/key-store=keystore:generate-key-pair(alias=user,algorithm=RSA,key-size=1024,validity=365,distinguished-name="CN=localhost")
+/subsystem=elytron/key-store=keystore:store()
+----
+
+Now we create a credential store and secret key under the Elytron subsystem, with the name `credstore`, and secret key alias `key`
+[source]
+----
+/subsystem=elytron/secret-key-credential-store=credstore:add(path=mycredstore.cs, relative-to=jboss.server.config.dir, create=true, populate=true)
```

Comment:
Might be worth mentioning what the populate attribute does (since this is what creates the secret key with alias `key`).

Reply:
Added